src/cpx_genericcallbacks.jl

```diff
@@ -0,0 +1,260 @@
+                    callback_data.context, cnt, ind .-= 1, val, obj, strat)
+    # The following codes should be added back when the error CPLEX 1003 is resolved.
+    # if return_status != 0
+    #     throw(CplexError(callback_data.model.env, return_status))
+    # end
+    # return return_status
+    return Cint(0)
+end
+
+function cbaddusercuts(callback_data::CallbackContext, rcnt::Int, nzcnt::Int, rhs::Cdouble, sense::Char, rmatbeg::Vector{Int}, rmatind::Vector{Int}, rmatval::Vector{Cdouble}, purgeable::Int, lcl::Int)
+    return_status = @cpx_ccall(callbackaddusercuts,
+                    Cint,
+                    (Ptr{Cvoid}, Cint, Cint, Ref{Cdouble}, Ptr{UInt8}, Ptr{Cint}, Ptr{Cint}, Ptr{Cdouble}, Ptr{Cint}, Ptr{Cint}),
+                    callback_data.context, rcnt, nzcnt, rhs, string(sense), rmatbeg .-= 1, rmatind .-= 1, rmatval, purgeable, lcl)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+# Check is the callback for the context CPX_CALLBACKCONTEXT_CANDIDATE is invoked by a feasible integer point
+function cbcandidateispoint(callback_data::CallbackContext, is_point::Int)
+    return_status = @cpx_ccall(callbackcandidateispoint,
+                    Cint,
+                    (Ptr{Cvoid}, Ptr{Cint}),
+                    callback_data.context, is_point)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+# Check is the callback for the context CPX_CALLBACKCONTEXT_CANDIDATE is invoked by an unbounded relaxation
+function cbcandidateisray(callback_data::CallbackContext, is_ray::Int)
+    return_status = @cpx_ccall(callbackcandidateisray,
+                    Cint,
+                    (Ptr{Cvoid}, Ptr{Cint}),
+                    callback_data.context, is_ray)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+function cbgetcandidatepoint(callback_data::CallbackContext, x::Vector{Cdouble}, begin_index::Int, end_index::Int, obj::Cdouble)
+    return_status = @cpx_ccall(callbackgetcandidatepoint,
+                    Cint,
+                    (Ptr{Cvoid}, Ptr{Cdouble}, Cint, Cint, Ref{Cdouble}),
+                    callback_data.context, x, begin_index - 1, end_index - 1, obj)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+function cbgetcandidateray(callback_data::CallbackContext, x::Vector{Cdouble}, begin_index::Int, end_index::Int)
+    return_status = @cpx_ccall(callbackgetcandidateray,
+                Cint,
+                (Ptr{Cvoid}, Ptr{Cdouble}, Cint, Cint),
+                callback_data.context, x, begin_index - 1, end_index - 1)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+function cbgetfunc(model::Model, context_id::Clong, callback_func::Function)
```

Comment:
I'm not sure this function does what you think it does?

https://www.ibm.com/support/knowledgecenter/en/SSSA5P_12.8.0/ilog.odms.cplex.help/refcallablelibrary/cpxapi/callbackgetfunc.html

Probably best to remove for now. I'm not sure how to wrap it, and I can't see there being much demand for it.

Reply:
I'll comment it out. Since it does not seem to be a priority I'll list it as a to-do and fix it some other time.